src/gui/symbology-ng/qgsellipsesymbollayerv2widget.cpp

```diff
@@ -109,6 +109,45 @@ void QgsEllipseSymbolLayerV2Widget::setSymbolLayer( QgsSymbolLayerV2* layer )
   mHorizontalAnchorComboBox->setCurrentIndex( mLayer->horizontalAnchorPoint() );
   mVerticalAnchorComboBox->setCurrentIndex( mLayer->verticalAnchorPoint() );
   blockComboSignals( false );
+
+  mDataDefinedPropertyButtons.clear();
+
+#define QGS_REGISTER_DATA_DEFINED_BUTTON( button, name, type, description )\
+  {\
+    QgsDataDefined dd( mLayer->dataDefinedProperty( name ) );\
+    button->init( mVectorLayer, &dd, QgsDataDefinedButton::type, tr( description ) );\
+    mDataDefinedPropertyButtons.insert( name, button );\
+  }
+  QGS_REGISTER_DATA_DEFINED_BUTTON( mSymbolWidthDDBtn, "width", Double, "" )
+  QGS_REGISTER_DATA_DEFINED_BUTTON( mSymbolHeightDDBtn, "height", Double, "" )
+  QGS_REGISTER_DATA_DEFINED_BUTTON( mRotationDDBtn, "rotation", Double, "" )
+  QGS_REGISTER_DATA_DEFINED_BUTTON( mOutlineWidthDDBtn, "outline_width", Double, "" )
+  QGS_REGISTER_DATA_DEFINED_BUTTON( mFillColorDDBtn, "fill_color", String, "'red,green,blue,alpha' e.g. '255,0,0,255'" )
+  QGS_REGISTER_DATA_DEFINED_BUTTON( mBorderColorDDBtn, "outline_color", String, "'red,green,blue,alpha' e.g. '255,0,0,255'" )
+  QGS_REGISTER_DATA_DEFINED_BUTTON( mShapeDDBtn, "symbol_name", String, "'circle', 'rectangle', 'cross', 'triangle'" )
+  QGS_REGISTER_DATA_DEFINED_BUTTON( mOffsetDDBtn, "offset", String, "'x,y' e.g. '2,3.5'" )
+  QGS_REGISTER_DATA_DEFINED_BUTTON( mHorizontalAnchorDDBtn, "horizontal_anchor_point", String, "'left', 'center' or 'right'" )
+  QGS_REGISTER_DATA_DEFINED_BUTTON( mVerticalAnchorDDBtn, "vertical_anchor_point", String, "'top', 'center' or 'bottom'" )  
+#undef QGS_REGISTER_DATA_DEFINED_BUTTON
+
+  for (QMap< QString, QgsDataDefinedButton* >::const_iterator it 
+          = mDataDefinedPropertyButtons.begin(); it != mDataDefinedPropertyButtons.end(); ++it )
+  {
+    connect( it.value(), SIGNAL( dataDefinedChanged( const QString& ) ), this, SLOT( updateDataDefinedProperty() ) );
+    connect( it.value(), SIGNAL( dataDefinedActivated( bool ) ), this, SLOT( updateDataDefinedProperty() ) );
+  }
+}
+
+void QgsEllipseSymbolLayerV2Widget::updateDataDefinedProperty()
```

Comment:
@vmora This should be in QgsSymbolLayerV2Widget to avoid the code duplication (it's implemented multiple times by this PR)